sycl/include/CL/sycl/types.hpp

```diff
@@ -264,12 +264,65 @@ detail::enable_if_t<is_float_to_int<T, R>::value, R> convertImpl(T Value) {
     assert(!"Unsupported rounding mode!");
     return static_cast<R>(Value);
   };
+}
 #else
-  // TODO implement device side conversion.
+
+template <typename T, typename R, rounding_mode roundingMode>
+detail::enable_if_t<!std::is_same<T, R>::value &&
+                        (is_int_to_int<T, R>::value &&
+                         !(is_int_to_from_uint<T, R>::value) ||
```

Comment:
This line seem cause build error, please double-check

Reply:
I forgot to delete it.